scripts/builtin/pcaPredict.dml

```diff
@@ -0,0 +1,49 @@
+#-------------------------------------------------------------
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+#-------------------------------------------------------------
+
+# Principal Component Analysis (PCA) for dimensionality reduction prediciton
+#
+# This method is used to predict on data, which the PCA was not trained on. To validate how good
+# The PCA is, and to apply in production. 
+#
+# ---------------------------------------------------------------------------------------------
+# NAME          TYPE    DEFAULT       MEANING
+# ---------------------------------------------------------------------------------------------
+# X             Matrix  ---           Input feature matrix
+# Centering     Matrix  empty matrix  The column means of the PCA model, subtracted to construct the PCA
+# ScaleFactor   Matrix  empty matrix  The scaling of each dimension in the PCA model
+# ---------------------------------------------------------------------------------------------
+# Y             Matrix  ---           Output feature matrix dimensionally reduced by PCA
+# ---------------------------------------------------------------------------------------------
+
+m_pcaPredict = function(Matrix[Double] X, Matrix[Double] Clusters, Matrix[Double] Centering,  Matrix[Double] ScaleFactor)
```

Comment:
Can we reconsider the name of this built-in? As from the word predict we usually refer to some class or value prediction but here you are somehow transforming the data using pre-computed dominant vectors. I would suggest pca_tranform() it would be consistent with PCA implementation in Scikit-learn.

Reply:
sure transform works as well! :+1:

Reply:
but i would refrain from the underscore. 
and we have to agree on that general syntax in our interface
are we using camel case "java syntax" or "google syntax" underscore?

Reply:
I prefer the camel case. For now, I see both styles in our built-in package but the camel case is dominant. So go for this.